PSModules/CompositeRegistrationFunctions.psm1

```diff
@@ -0,0 +1,500 @@
+        TopNavigationText = $Path.TopNavigationText
+        TopNavigationOrder = $Path.TopNavigationOrder
+        Layout = $Path.Layout
+        IsOnline = $Path.IsOnline
+        OfflineHtml = $Path.OfflineHtml
+        PhaseBannerUrl = $Path.PhaseBannerHtml
+        ExternalUrl = $Path.ExternalUrl
+        SitemapURL = $Path.SitemapUrl
+        RobotsURL = $Path.RobotsUrl
+    }
+
+    $requestBodyText = $requestBody | ConvertTo-Json
+
+    $finalUrl = "$($script:PathApiUrl)/paths"
+
+    return Invoke-CompositeApiRegistrationRequest -Url $finalUrl -Method Post -RequestBody $requestBodyText
+}
+
+function New-RegionRegistration
+{
+<#
+.SYNOPSIS
+Creates a new region registration
+
+.DESCRIPTION
+Creates a new region registration
+
+.PARAMETER Path
+The path that the region belongs to
+
+.PARAMETER Region
+An object containing the region to be created
+
+.EXAMPLE
+
+$regionObject = @{
+    PageRegion = 1
+}
+
+New-RegionRegistration -Path somePath -Region $regionObject
+#> 
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Path,
+        [Parameter(Mandatory=$true)]
+        [object] $Region
+    )
+
+    if($null -eq $Region.PageRegion) { throw "PageRegion is not set for a region on path $Path."}
+    if($null -eq $Region.HealthCheckRequired) { 
+        $Region | Add-Member -NotePropertyName HealthCheckRequired -NotePropertyValue $true 
+    }
+
+    if($null -eq $Region.IsHealthy) { 
+        $Region | Add-Member -NotePropertyName IsHealthy -NotePropertyValue $true 
+    }
+
+    $requestBody = @{
+        Path = $Path
+        PageRegion = $Region.PageRegion
+        IsHealthy = $Region.IsHealthy
+        RegionEndpoint = $Region.RegionEndpoint
+        HeathCheckRequired = $Region.HealthCheckRequired
+        OfflineHTML = $Region.OfflineHtml
+    }
+
+    $requestBodyText = $requestBody | ConvertTo-Json
+
+    $finalUrl = "$($script:RegionApiUrl)/paths/$Path/regions"
+
+    return Invoke-CompositeApiRegistrationRequest -Url $finalUrl -Method Post -RequestBody $requestBodyText    
+}
+
+function Update-PathRegistration
+{
+<#
+.SYNOPSIS
+Patches a existing path registration
+
+.DESCRIPTION
+Patches a existing path registration.  Only fields that are passed in will be changed!
+
+.PARAMETER Path
+The path of the path registration to update.
+
+.PARAMETER ItemsToUpdate
+A dictionary containing items to update.
+Note that this *must* contain at least the name of the path registration itself.
+
+.EXAMPLE
+$itemsToUpdate = @{
+    Path = "somePath"
+    Layout = 1
+}
+
+Update-PathRegistration -Path somePath -ItemsToUpdate $itemsToUpdate
+#> 
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Path,
+        [Parameter(Mandatory=$true)]
+        [object] $ItemsToUpdate
+    )
+ 
+    $itemsToPatch = [array] @()
+
+    foreach($item in $ItemsToUpdate.Keys) {
+        if($item -eq "Path") { continue }
+
+        $itemsToPatch += @{
+            "op" = "Replace"
+            "path" = "/$($item)"
```

Comment:
This could do with a comment to explain that the property path of $itemsToPatch isn't the same as the parameter Path and the property Path of Items to update.  Some context would be useful as well, not sure I'd realise it was a patch object if I hadn't overheard the convo earlier